eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -974,6 +987,20 @@ submitFailingTx ::
   ImpTestM era ()
 submitFailingTx tx expectedFailure = trySubmitTx tx >>= (`shouldBeLeftExpr` expectedFailure)
 
+submitFailingTxWithSubset ::
+  ( HasCallStack
+  , ShelleyEraImp era
+  ) =>
+  Tx era ->
+  NonEmpty (PredicateFailure (EraRule "LEDGER" era)) ->
+  ImpTestM era ()
+submitFailingTxWithSubset tx expectedFailuresSubset =
+  trySubmitTx tx
+    >>= expectLeftExpr
+    >>= ( `shouldSatisfy`
+            (\fs -> all (`elem` fs) expectedFailuresSubset)
+        )
```

Comment:
I am not sure I like this approach of testing only presence of some of the failures. 
By checking the exact sequence of predicate failures we are testing the exact behavior of the rule. If a rule has a slightly different outcome from one era to another then we should have separate tests for such rule.

I understand that it is easier to solve it this way, but I believe this is a dangerous slope. Moreover, we are translating some of the tests that checked exact behavior, thus in this PR we are making our test suite weaker, rather than better. 

More importantly, we have all of the tooling necessary to deal with this sort of discrepancies between eras in a more accurate way. See my suggestion on the `ExtraneousScriptWitnessesUTXOW` test for an example.

So, could you please remove this `submitFailingTxWithSubset` and switch back to `submitFailingTx` instead

Reply:
I agree.  Removed `submitFailingTxWithSubset`  and actually it wasn't too bad to do without it, thank you.